### PR TITLE
Refactor authCallbackLti

### PR DESCRIPTION
**Pull request by nwalters512**
This was done primarily with the goal of fixing broken nonce checking; see inline comment for more details. However, that was easiest to do by just rewriting all the code to use `async`/`await`, which I did. I also took the opportunity to improve the HTTP status codes used here.

Best reviewed with whitespace changes hidden.

### CODE REVIEW

File: apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js [hunk 1]
```diff
@@ -15,202 +13,168 @@ const { config } = require('../../lib/config');
...
-      var nonceKey = 'authCallbackLti:' + parameters.oauth_timestamp + ':' + parameters.oauth_nonce;
-      util.callbackify(cache.get)(nonceKey, (err, val) => {
-        if (ERR(err, next)) return;
-        if (val) {
-          nonceReused = true;
-        } else {
-          cache.set(nonceKey, true, timeTolerance * 1000);
-        }
-      });
-      if (nonceReused) return next(error.make(500, 'Invalid nonce reuse'));
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js:90]: This is what I was aiming to fix. The code as written is nonsensical: it starts an async task to fetch the nonce from the cache but then synchronously reads `nonceReused`, which will always be `false` at that point.

File: apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js [hunk 2]
```diff
@@ -15,202 +13,168 @@ const { config } = require('../../lib/config');
...
+      resource_link_id: parameters.resource_link_id,
+      resource_link_title: parameters.resource_link_title || '',
+      resource_link_description: parameters.resource_link_description || '',
+    });
+
+    // Do we have an assessment linked to this resource_link_id?
+    if (linkResult.rows[0].assessment_id) {
+      if ('lis_result_sourcedid' in parameters) {
+        // Save outcomes here
+        await sqldb.queryAsync(sql.upsert_outcome, {
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js:144]: The previous behaviour had this done in the background. Just double-checking that you're ok with awaiting for this update before redirecting to the assessment.

**Review by jonatanschroeder — COMMENTED** [review]
I don't feel confident in this code enough to properly review it, so just adding a note of something I noticed.

File: apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/authCallbackLti/authCallbackLti.js:144]: This is just a database query, there's nothing that should take any noticeable amount of time here, so I'm very comfortable doing this inline.

**Review by mylesw27 — APPROVED** [review]

**Pull request merged**